src/main/java/de/samply/share/broker/filter/BasicAuthenticationFilter.java

```diff
@@ -0,0 +1,84 @@
+package de.samply.share.broker.filter;
+
+import de.samply.share.common.utils.ProjectInfo;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.lang.reflect.AnnotatedElement;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.List;
+import javax.annotation.Priority;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.Provider;
+import org.apache.http.HttpHeaders;
+
+@BasicAuthSecure
+@Provider
+@Priority(Priorities.AUTHENTICATION)
+public class BasicAuthenticationFilter implements ContainerRequestFilter {
+
+  private static final String AUTHENTICATION_SCHEME = "Basic";
+  @Context
+  private ResourceInfo resourceInfo;
+
+  @Override
+  public void filter(ContainerRequestContext requestContext) throws IOException {
+    String authHeader = requestContext.getHeaderString(HttpHeaders.AUTHORIZATION);
+    if (authHeader == null || !authHeader.startsWith(AUTHENTICATION_SCHEME)) {
+      requestContext.abortWith(Response.status(Response.Status.UNAUTHORIZED)
+          .header("WWW-Authenticate", AUTHENTICATION_SCHEME).build());
+      return;
+    }
+    Method resourceMethod = resourceInfo.getResourceMethod();
+    List<BasicAuthPermission> methodPermissions = extractPermissions(resourceMethod);
+    if (!checkPermissions(methodPermissions, authHeader)) {
+      requestContext.abortWith(Response.status(Response.Status.UNAUTHORIZED).build());
+    }
+  }
+
+  private List<BasicAuthPermission> extractPermissions(AnnotatedElement annotatedElement) {
+    if (annotatedElement == null) {
+      return new ArrayList<>();
+    } else {
+      BasicAuthSecure secured = annotatedElement.getAnnotation(BasicAuthSecure.class);
+      if (secured == null) {
+        return new ArrayList<>();
+      } else {
+        BasicAuthPermission[] allowedPermissions = secured.value();
+        return Arrays.asList(allowedPermissions);
+      }
+    }
+  }
+
+  private boolean checkPermissions(List<BasicAuthPermission> allowedAccessPermissions,
+      String authHeader) throws UnsupportedEncodingException {
+    String[] tokens = (new String(Base64.getDecoder().decode(authHeader.split(" ")[1]),
+        "UTF-8")).split(":");
+    final String username = tokens[0];
+    final String password = tokens[1];
+    String usernameConfig = "";
+    String passwordConfig = "";
+    for (BasicAuthPermission basicAuthPermission : allowedAccessPermissions) {
+      if (BasicAuthPermission.STRUCTURED_QUERY.equals(
+          basicAuthPermission)) {
+        usernameConfig = ProjectInfo.INSTANCE.getConfig().getProperty("structured.query.username");
```

Comment:
please put the config name in the BasicAuthPermission Enum, so that you don't need an if statement here.